components/driver/twai.c

```diff
@@ -446,7 +449,8 @@ esp_err_t twai_driver_install(const twai_general_config_t *g_config, const twai_
 
     //Allocate GPIO and Interrupts
     twai_configure_gpio(g_config->tx_io, g_config->rx_io, g_config->clkout_io, g_config->bus_off_io);
-    ESP_ERROR_CHECK(esp_intr_alloc(ETS_TWAI_INTR_SOURCE, g_config->intr_flags, twai_intr_handler_main, NULL, &p_twai_obj->isr_handle));
+    ret = esp_intr_alloc(ETS_TWAI_INTR_SOURCE, g_config->intr_flags, twai_intr_handler_main, NULL, &p_twai_obj->isr_handle);
```

Comment:
Prefer if we

- Allocated interrupts immediately after allocating the driver object. This way, if any resource allocation fails, we can return early
- Added the `ESP_INTR_FLAG_INTRDISABLED` flag so that an allocated interrupt remains disabled until we explicitly enable it using `esp_intr_enable()`

So something like...

```suggestion
    //Create a TWAI object (including queues and semaphores)
    p_twai_obj_dummy = twai_alloc_driver_obj(g_config->tx_queue_len, g_config->rx_queue_len);
    TWAI_CHECK(p_twai_obj_dummy != NULL, ESP_ERR_NO_MEM);
    //Allocate interrupt for TWAI driver
    intr_handle_t isr_hdl;
    ret = esp_intr_alloc(ETS_TWAI_INTR_SOURCE, g_config->intr_flags | ESP_INTR_FLAG_INTRDISABLED, twai_intr_handler_main, NULL, &isr_hdl);
    if (ret != ESP_OK) {
        ESP_LOGE(TWAI_TAG, "Interrupt allocation failed");
        goto intr_alloc_err;
    }
```